prov/efa/src/efa_cq.c

```diff
@@ -50,42 +49,53 @@ static uint64_t efa_cq_wc_to_fi_flags(struct efa_wc *wc)
 	}
 }
 
+/* @brief Unlike ibv_poll_cq, wide completion APIs do not write to ibv_wc struct. We need to do this manually.
+ * This is inspired by rdma-core, i.e. `efa_process_cqe` and `efa_process_ex_cqe`.
+ */
+static inline void efa_cq_cw_from_ibv_cq_ex_unsafe(struct ibv_cq_ex *cq, struct ibv_wc *wc) {
+	wc->status = cq->status;
+	wc->vendor_err = ibv_wc_read_vendor_err(cq);
+	wc->wc_flags = ibv_wc_read_wc_flags(cq);
+	wc->qp_num = ibv_wc_read_qp_num(cq);
+	wc->opcode = ibv_wc_read_opcode(cq);
+	wc->byte_len = ibv_wc_read_byte_len(cq);
+	wc->src_qp = ibv_wc_read_src_qp(cq);
+	wc->sl = ibv_wc_read_sl(cq);
+	wc->slid = ibv_wc_read_slid(cq);
+	wc->imm_data = ibv_wc_read_imm_data(cq);
+	wc->wr_id = cq->wr_id;
+}
+
 ssize_t efa_cq_readerr(struct fid_cq *cq_fid, struct fi_cq_err_entry *entry,
```

Comment:
I do not have a test against this function. Ideas are welcome.

Reply:
I think we can construct a unit test through cmocka.

Some like:

```
   will_return(ibv_start_poll, err_code);

   ret = efa_cq_readfrom();
   assert (ret == FI_EAVAIL);
   ret = efa_cq_readerr(cq, entry);
   assert(entry.err == ibv_err);
```

Reply:
That looks good. I will try it out. But I'm not familiar with unit testing C code. I might have to do this separately from this CR for time sake. Are you ok with this?